shared/src/main/scala/tastyquery/Contexts.scala

```diff
@@ -37,7 +37,25 @@ object Contexts {
     def withFile(filename: String): FileContext =
       new FileContext(defn, defn.RootPackage, filename, classloader)
 
-    def createPackageSymbolIfNew(name: TermName, owner: Symbol): PackageClassSymbol = {
+    def createClassSymbolIfNew(name: Name, owner: DeclaringSymbol): ClassSymbol =
+      owner.getDecl(name) match {
+        case None =>
+          val cls = ClassSymbolFactory.createSymbol(name, owner)
+          owner.addDecl(cls)
+          cls
+        case Some(clazz) => ClassSymbolFactory.castSymbol(clazz)
```

Comment:
In this case we need to rethink the way that classpath scanning works: currently we only traverse tasty files (which may make top level classes, e.g. `Foo` in say package `example`), then when we need the symbol for a Java dependency in `example` package, we then scan the classpath and create top level symbols - which will see `example.Foo` for the second time

Reply:
I had avoided doing full classpath scanning from the beginning, because full dependency symbols would be unnecessary if we have a mode where someone just wants to see trees, with no semantic information

Reply:
> I had avoided doing full classpath scanning from the beginning, because full dependency symbols would be unnecessary if we have a mode where someone just wants to see trees

That sounds like premature optimization to me. *Scanning* a package in a classpath for top-level entries should never be so expensive that we would gain something substantial by only focusing on the ones with a `.tasty` file attached.

Reply:
so to make this change, we will scan top level symbols first, then TreeUnpickler will need to be rewritten to make creating a classymbol a hard error when the symbol already exists - and then we will need to create a TreeUnpickler with the top level symbols in its constructor